From a5925cfc862923544d4d2b4dc5264836e2cd1030 Mon Sep 17 00:00:00 2001
From: Nick Korotysh <kolchaprogrammer@list.ru>
Date: Thu, 21 Apr 2022 01:02:32 +0300
Subject: [PATCH] fixed pkg-config file libraries list generation

do not append '-l' to anything starting with '-'

previously cmake-generated pkg-config file contained next line:

  Libs: -L${libdir} -ltorrent-rasterbar -l-pthread -lssl -lcrypto

(note '-l' before '-pthread', even without space)

the same line with this fix included is correct:

  Libs: -L${libdir} -ltorrent-rasterbar -pthread -lssl -lcrypto
---
 cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in b/cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in
index d4cb81db69..6adea52ac5 100644
--- a/cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in
+++ b/cmake/Modules/GeneratePkgConfig/generate-pkg-config.cmake.in
@@ -37,7 +37,12 @@ function(split_library_dirs _libraries _base_library_dir _library_dirs_var _libr
 endfunction()
 
 split_library_dirs("${_TARGET_INTERFACE_LINK_LIBRARIES}" "${CMAKE_INSTALL_PREFIX}/${_INSTALL_LIBDIR}" _lib_dirs _library_names)
+set(_linker_options "${_library_names}")
+list(FILTER _linker_options INCLUDE REGEX "^-.*")
+list(FILTER _library_names EXCLUDE REGEX "^-.*")
 cmake_list_to_pkg_config(_libs "${_library_names}" "-l")
+list(JOIN _linker_options " " _linker_options)
+string(JOIN " " _libs "${_linker_options}" "${_libs}")
 list(LENGTH _lib_dirs _additional_libdirs_count)
 if (_additional_libdirs_count GREATER 0)
 	cmake_list_to_pkg_config(_additional_libdirs "${_lib_dirs}" "-L")
